compatibility_server/pip_checker.py

```diff
@@ -197,20 +199,42 @@ def __init__(self,
         self._tmp_path = tmp_path
         self._output_directory = None
         self._clean = clean
+        self._docker_client = docker.from_env()
```

Comment:
The `_container` is needed for most of the functions in this class, so I'd like to create it in the constructor rather that making a `self._container` in the `run` function, which will also work but looks a little strange.

Reply:
It's up to you. I try to avoid using external dependencies in constructors because it can make testing difficult i.e. because you can't even create the object without invoking the external dependency.

But I can see how your way is cleaner.

Reply:
Sounds good! Moved the docker stuff to `run`.